### ISSUE TITLE
findOne using Not and relation 

### ISSUE THREAD

**Issue by apellizzn** [typeorm/typeorm#2454]
**Issue type:**

[ x] question
[ ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[x ] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

Hello, i am performing a `findOne` while trying to filter for a relation
```
const { Not } = require('typeorm')
...
const email: 

userRepo.findOne({
 where: { email: 'me@example.com', organization: Not(myOrganization) } 
})
```
Is this the correct way doing it ? Seems like it does not work :/

**Comment by pleerock** [comment]
syntax you are proposal is available in `@next` via #2100 and will be part of 0.3.0 release:

```ts
userRepo.findOne({
 where: { email: 'me@example.com', organization: { id: Not(myOrganization)  }} 
})
```

For now you need to use solution you already found.

**Comment by apellizzn** [comment]
Ok, thank you
